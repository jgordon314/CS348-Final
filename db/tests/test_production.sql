SOURCE prod/load_prod.sql;

-- R6
SELECT id, username FROM Users WHERE username = 'mwgrk' AND password = '13104';
INSERT INTO Users (username, password) VALUES ('billnye', 'the5c1enceguy!');
SELECT id, username FROM Users WHERE username = 'billnye' AND password = 'the5c1enceguy!';

-- R7
UPDATE CharacterSpellList csi SET activations = activations + 1 WHERE csi.spell_id = 1 AND csi.character_id = 1;
UPDATE CharacterInventory ci JOIN Items i ON ci.item_id = i.id SET ci.activation_count = ci.activation_count + 1 WHERE ci.character_id = 1 AND i.ability_id = 1;

-- R8
(
    SELECT sd.*, 1 AS multiplier
    FROM SkillDeltas sd, Characters c
    WHERE c.id = 1 AND sd.id = c.base_stat_id
) UNION ALL (
    SELECT sd.*, csl.activations AS multiplier
    FROM SkillDeltas sd, Spells s, CharacterSpellList csl, Characters c
    WHERE c.id = 1 AND s.id = csl.spell_id AND csl.character_id = c.id AND sd.id = s.skill_delta_id AND csl.activations > 0
) UNION ALL (
    SELECT sd.*, ca.activation_count AS multiplier
    FROM SkillDeltas sd, Abilities a, CharacterAbilities ca, Characters c
    WHERE c.id = 1 AND a.id = ca.ability_id AND ca.character_id = c.id AND sd.id = a.skill_delta_id AND ca.activation_count > 0
) UNION ALL (
    SELECT sd.*, ci.activation_count AS multiplier
    FROM SkillDeltas sd, Abilities a, Items i, CharacterInventory ci, Characters c
    WHERE c.id = 1 AND i.id = ci.item_id AND ci.character_id = c.id AND a.id = i.ability_id AND sd.id = a.skill_delta_id AND ci.activation_count > 0
);

-- R9
UPDATE CharacterInventory SET activation_count = 0 WHERE character_id = 1 AND item_id = 1;

-- R10
SELECT a.* FROM Abilities a LEFT JOIN CharacterAbilities ca ON a.id = ca.ability_id AND ca.character_id = 1 WHERE ca.character_id IS NULL LIMIT 5;
SELECT i.* FROM Items i LEFT JOIN CharacterInventory ci ON i.id = ci.item_id AND ci.character_id = 1 WHERE ci.character_id IS NULL LIMIT 5;
SELECT s.* FROM Spells s LEFT JOIN CharacterSpellList csl ON s.id = csl.spell_id AND csl.character_id = 1 WHERE csl.character_id IS NULL LIMIT 5;

-- R11
START TRANSACTION; 
    INSERT INTO SkillDeltas (
    armor_class, current_health, max_health,
    strength, dexterity, constitution, intelligence, wisdom, charisma,
    acrobatics, animal_handling, arcana, athletics, deception,
    history, insight, intimidation, investigation, medicine,
    nature, perception, performance, persuasion, religion,
    sleight_of_hand, stealth, survival
    ) VALUES (
    3, 12, 20,
    10, 14, 12, 11, 13, 9,
    2, 1, 3, 4, 0,
    2, 3, 1, 2, 1,
    2, 4, 0, 1, 3,
    2, 5, 2
    );
    INSERT INTO Spells (name, level, description, skill_delta_id, components) VALUES ('Firebolt', 0, 'A bolt of fire deals 1d10 damage.', LAST_INSERT_ID(), '');
COMMIT;
SELECT name, level, description FROM Spells WHERE id = LAST_INSERT_ID();

START TRANSACTION; 
    INSERT INTO SkillDeltas (
    armor_class, current_health, max_health,
    strength, dexterity, constitution, intelligence, wisdom, charisma,
    acrobatics, animal_handling, arcana, athletics, deception,
    history, insight, intimidation, investigation, medicine,
    nature, perception, performance, persuasion, religion,
    sleight_of_hand, stealth, survival
    ) VALUES (
    2, 8, 15,
    9, 13, 11, 12, 10, 8,
    1, 0, 4, 2, 3,
    1, 2, 0, 3, 1,
    2, 3, 1, 2, 1,
    2, 3, 2
    );
    INSERT INTO Abilities (name, type, description, skill_delta_id) VALUES ('Heal', 'action', 'Restore 2d4 + 2 HP', LAST_INSERT_ID());
    INSERT INTO Items (name, description, ability_id) VALUES ('Healing Potion', 'Effect: Restore 2d4 + 2 HP', LAST_INSERT_ID());
COMMIT;
SELECT Items.name AS item_name, Items.description AS item_description,
Abilities.name AS ability_name, Abilities.type AS ability_type, Abilities.description AS ability_description
FROM Items, Abilities WHERE Items.id = LAST_INSERT_ID() AND Items.id = Abilities.id;

START TRANSACTION; 
    INSERT INTO SkillDeltas (
    armor_class, current_health, max_health,
    Strength, Dexterity, Constitution, Intelligence, Wisdom, Charisma,
    Acrobatics, Animal_Handling, Arcana, Athletics, Deception,
    History, Insight, Intimidation, Investigation, Medicine,
    Nature, Perception, Performance, Persuasion, Religion,
    Sleight_of_Hand, Stealth, Survival
    ) VALUES (
    5, 20, 25,
    15, 12, 13, 11, 10, 14,
    3, 2, 1, 5, 2,
    4, 1, 2, 3, 2,
    1, 3, 2, 3, 4,
    1, 1, 3
    );
    INSERT INTO Abilities (name, type, description, skill_delta_id) VALUES ('Rage', 2, 'Enter a frenzied rage for 1 minute.', LAST_INSERT_ID());
COMMIT;
SELECT name, type, description FROM Abilities WHERE id = LAST_INSERT_ID();

-- R12
CREATE VIEW CharacterBaseCharismas AS
SELECT Characters.id AS character_id, Characters.name AS character_name, SkillDeltas.charisma AS character_base_charisma
FROM Characters, SkillDeltas WHERE Characters.base_stat_id = SkillDeltas.id;

SELECT * FROM CharacterBaseCharismas WHERE character_id = 4;
UPDATE SkillDeltas SET charisma = charisma + 3 WHERE id = (SELECT base_stat_id FROM Characters WHERE id = 4);
SELECT * FROM CharacterBaseCharismas WHERE character_id = 4;

DROP VIEW CharacterBaseCharismas;

-- R13
CREATE INDEX CharacterUserId ON Characters(user_id);
SELECT * FROM Characters WHERE user_id = 2; 

-- R14
START TRANSACTION;
    UPDATE CharacterInventory SET activation_count = 0 WHERE character_id = 1 AND activation_count > 0;
    UPDATE CharacterAbilities SET activation_count = 0, available_uses = max_uses WHERE character_id = 1;
    UPDATE CharacterSpellList SET activations = 0 WHERE character_id = 1;
    UPDATE SkillDeltas s JOIN Characters c ON c.base_stat_id = s.id SET s.current_health = s.max_health WHERE c.id = 1;
COMMIT;

-- R15
CREATE UNIQUE INDEX CharacterId ON Characters(id);
CREATE INDEX CharacterSpellListCharacterId ON CharacterSpellList(character_id);
CREATE INDEX CharacterInventoryId ON CharacterInventory(character_id);
CREATE INDEX CharacterAbilitiesId ON CharacterAbilities(character_id);

INSERT INTO CharacterSpellList (character_id, spell_id, activations) VALUES (2, 11, 3);
SELECT Characters.name AS charcter_name, Spells.name AS spell_name, CharacterSpellList.activations AS activations
FROM Characters, Spells, CharacterSpellList
WHERE Characters.id = 2 AND Characters.id = CharacterSpellList.character_id AND Spells.id = CharacterSpellList.spell_id LIMIT 5;

INSERT INTO CharacterInventory (character_id, item_id, quantity) VALUES (4, 5, 2);
SELECT Characters.name AS charcter_name, Items.name AS item_name, CharacterInventory.quantity AS item_quantity
FROM Characters, Items, CharacterInventory
WHERE Characters.id = 3 AND Characters.id = CharacterInventory.character_id AND Items.id = CharacterInventory.item_id LIMIT 5;

INSERT INTO CharacterAbilities (character_id, ability_id) VALUES (4, 8);
SELECT Characters.name AS charcter_name, Abilities.name AS ability_name
FROM Characters, Abilities, CharacterAbilities
WHERE Characters.id = 4 AND Characters.id = CharacterAbilities.character_id AND Abilities.id = CharacterAbilities.ability_id LIMIT 5;
