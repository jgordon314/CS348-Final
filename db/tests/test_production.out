MySQL [changeme]> source db/tests/test_production.sql
ERROR at line 1 in file: 'db/tests/test_production.sql': Failed to open file 'prod/load_prod.sql', error: 2
+----+----------+
| id | username |
+----+----------+
|  1 | mwgrk    |
+----+----------+
1 row in set (0.001 sec)

Query OK, 1 row affected (0.008 sec)

+----+----------+
| id | username |
+----+----------+
| 11 | billnye  |
+----+----------+
1 row in set (0.001 sec)

Query OK, 0 rows affected (0.000 sec)
Rows matched: 0  Changed: 0  Warnings: 0

Query OK, 0 rows affected (0.000 sec)
Rows matched: 0  Changed: 0  Warnings: 0

+----+-------------+----------------+------------+----------+-----------+--------------+--------+----------+--------------+-----------+------------+-----------------+---------+--------+---------+---------------+--------+----------+-----------------+---------+----------+------------+----------+-----------+--------------+-------------+------------+------------+
| id | armor_class | current_health | max_health | strength | dexterity | intelligence | wisdom | charisma | constitution | athletics | acrobatics | sleight_of_hand | stealth | arcana | history | investigation | nature | religion | animal_handling | insight | medicine | perception | survival | deception | intimidation | performance | persuasion | multiplier |
+----+-------------+----------------+------------+----------+-----------+--------------+--------+----------+--------------+-----------+------------+-----------------+---------+--------+---------+---------------+--------+----------+-----------------+---------+----------+------------+----------+-----------+--------------+-------------+------------+------------+
|  1 |          18 |             45 |         45 |       14 |        10 |            8 |     12 |       10 |           14 |         5 |          0 |              -1 |       0 |      0 |       2 |             1 |      0 |        0 |               0 |       3 |        0 |          2 |        0 |         1 |            0 |           2 |          0 |          1 |
+----+-------------+----------------+------------+----------+-----------+--------------+--------+----------+--------------+-----------+------------+-----------------+---------+--------+---------+---------------+--------+----------+-----------------+---------+----------+------------+----------+-----------+--------------+-------------+------------+------------+
1 row in set (0.001 sec)

Query OK, 0 rows affected (0.000 sec)
Rows matched: 0  Changed: 0  Warnings: 0

+----+------------------+----------------------------------------------------------------------------------------------------------------+----------------+--------------+
| id | name             | description                                                                                                    | skill_delta_id | type         |
+----+------------------+----------------------------------------------------------------------------------------------------------------+----------------+--------------+
|  1 | Shield Wall      | You and adjacent allies gain a +2 bonus to armor class until the start of your next turn.                      |           NULL | reaction     |
|  2 | Frenzied Strike  | Make an additional weapon attack as a bonus action this turn.                                                  |           NULL | bonus-action |
|  3 | Cloak of Shadows | You become invisible until the start of your next turn, or until you attack or cast a spell.                   |           NULL | action       |
|  4 | Rallying Cry     | Choose a number of allies equal to your proficiency bonus. They gain temporary hit points equal to your level. |           NULL | action       |
|  5 | Divine Shield    | Gain a +1 bonus to AC and advantage on saving throws until the start of your next turn.                        |           NULL | action       |
+----+------------------+----------------------------------------------------------------------------------------------------------------+----------------+--------------+
5 rows in set (0.000 sec)

+----+----------------------+----------------------------------------------------------------------------------------------------------------+------------+
| id | name                 | description                                                                                                    | ability_id |
+----+----------------------+----------------------------------------------------------------------------------------------------------------+------------+
|  1 | Frostfire Blade      | A sword that flickers with icy blue flames. Deals fire and cold damage.                                        |         19 |
|  2 | Ring of Shadows      | A dark silver ring that grants the wearer greater stealth capabilities.                                        |         20 |
|  3 | Cloak of the Phoenix | A vibrant cloak that allows the wearer to burst into flames upon death, dealing fire damage to nearby enemies. |         21 |
|  4 | Amulet of Wisdom     | A golden amulet adorned with a sapphire, enhancing the wearer's wisdom and knowledge.                          |         22 |
|  5 | Boots of the Cat     | These soft, light boots grant incredible agility and reaction time.                                            |         23 |
+----+----------------------+----------------------------------------------------------------------------------------------------------------+------------+
5 rows in set (0.000 sec)

+----+----------------------+-------+-----------------------------------------------------------------------------------------------------------------------+----------+----------------+-----------------------+-------------+---------------------+
| id | name                 | level | description                                                                                                           | duration | skill_delta_id | casting_time          | spell_range | components          |
+----+----------------------+-------+-----------------------------------------------------------------------------------------------------------------------+----------+----------------+-----------------------+-------------+---------------------+
|  1 | Flame Lash           |     2 | A whip made of flames lashes out at a target, dealing fire damage and igniting them for further burns.                |        3 |           NULL | action                | 30ft        | V, S                |
|  2 | Spectral Shield      |     1 | A shimmering barrier appears, granting bonus armor class against the next attack.                                     |        1 |             31 | reaction with trigger | self        | V, S                |
|  3 | Whispers of the Mind |     3 | You invade the mind of a target, forcing them to make a wisdom saving throw or be incapacitated for the duration.     |        4 |           NULL | action                | 60ft        | V, S, M (a feather) |
|  4 | Mend Wounds          |     0 | Heals a small amount of damage to a target, knitting their wounds with magical energy.                                |        1 |             32 | action                | touch       | V, S                |
|  5 | Thunderous Step      |     2 | You teleport up to 30ft and leave a thunderous explosion at your original location, dealing damage to nearby enemies. |        0 |           NULL | bonus action          | self        | V                   |
+----+----------------------+-------+-----------------------------------------------------------------------------------------------------------------------+----------+----------------+-----------------------+-------------+---------------------+
5 rows in set (0.000 sec)

Query OK, 0 rows affected (0.000 sec)

Query OK, 1 row affected (0.000 sec)

Query OK, 1 row affected (0.000 sec)

Query OK, 0 rows affected (0.002 sec)

+----------+-------+-----------------------------------+
| name     | level | description                       |
+----------+-------+-----------------------------------+
| Firebolt |     0 | A bolt of fire deals 1d10 damage. |
+----------+-------+-----------------------------------+
1 row in set (0.001 sec)

Query OK, 0 rows affected (0.000 sec)

Query OK, 1 row affected (0.001 sec)

Query OK, 1 row affected (0.000 sec)

Query OK, 1 row affected (0.000 sec)

Query OK, 0 rows affected (0.002 sec)

+----------------+----------------------------+--------------+--------------+----------------------------------------------+
| item_name      | item_description           | ability_name | ability_type | ability_description                          |
+----------------+----------------------------+--------------+--------------+----------------------------------------------+
| Healing Potion | Effect: Restore 2d4 + 2 HP | Wind Dancer  | non-action   | You gain a +10 bonus to your movement speed. |
+----------------+----------------------------+--------------+--------------+----------------------------------------------+
1 row in set (0.000 sec)

Query OK, 0 rows affected (0.000 sec)

Query OK, 1 row affected (0.000 sec)

Query OK, 1 row affected (0.000 sec)

Query OK, 0 rows affected (0.001 sec)

+------+--------+-------------------------------------+
| name | type   | description                         |
+------+--------+-------------------------------------+
| Rage | action | Enter a frenzied rage for 1 minute. |
+------+--------+-------------------------------------+
1 row in set (0.001 sec)

Query OK, 0 rows affected (0.003 sec)

+--------------+------------------+-------------------------+
| character_id | character_name   | character_base_charisma |
+--------------+------------------+-------------------------+
|            4 | Xandril the Wise |                      16 |
+--------------+------------------+-------------------------+
1 row in set (0.001 sec)

Query OK, 1 row affected (0.002 sec)
Rows matched: 1  Changed: 1  Warnings: 0

+--------------+------------------+-------------------------+
| character_id | character_name   | character_base_charisma |
+--------------+------------------+-------------------------+
|            4 | Xandril the Wise |                      19 |
+--------------+------------------+-------------------------+
1 row in set (0.001 sec)

Query OK, 0 rows affected (0.002 sec)

Query OK, 0 rows affected (0.021 sec)
Records: 0  Duplicates: 0  Warnings: 0

+----+------------------+---------------------------------------------------------------------------+---------+--------------+
| id | name             | description                                                               | user_id | base_stat_id |
+----+------------------+---------------------------------------------------------------------------+---------+--------------+
|  2 | Elara Moonshadow | A 3rd-level Elf Rogue with a talent for stealth and subterfuge.           |       2 |            2 |
|  8 | Kara Dawnsinger  | A 3rd-level Half-Elf Cleric devoted to healing and supporting her allies. |       2 |            8 |
| 21 | Thalindra        | Level 5 Elven Sorcerer, skilled in elemental magic.                       |       2 |           21 |
+----+------------------+---------------------------------------------------------------------------+---------+--------------+
3 rows in set (0.000 sec)

Query OK, 0 rows affected (0.000 sec)

Query OK, 7 rows affected (0.001 sec)
Rows matched: 7  Changed: 7  Warnings: 0

Query OK, 3 rows affected (0.001 sec)
Rows matched: 4  Changed: 3  Warnings: 0

Query OK, 1 row affected (0.001 sec)
Rows matched: 3  Changed: 1  Warnings: 0

Query OK, 0 rows affected (0.000 sec)
Rows matched: 1  Changed: 0  Warnings: 0

Query OK, 0 rows affected (0.002 sec)

Query OK, 0 rows affected (0.020 sec)
Records: 0  Duplicates: 0  Warnings: 0

Query OK, 0 rows affected (0.014 sec)
Records: 0  Duplicates: 0  Warnings: 0

Query OK, 0 rows affected (0.015 sec)
Records: 0  Duplicates: 0  Warnings: 0

Query OK, 0 rows affected (0.015 sec)
Records: 0  Duplicates: 0  Warnings: 0

Query OK, 1 row affected (0.002 sec)

+------------------+----------------------+-------------+
| charcter_name    | spell_name           | activations |
+------------------+----------------------+-------------+
| Elara Moonshadow | Whispers of the Mind |           3 |
| Elara Moonshadow | Thunderous Step      |           0 |
| Elara Moonshadow | Frostbite Aura       |           1 |
| Elara Moonshadow | Aura of Clarity      |           0 |
| Elara Moonshadow | Empowered Strike     |           1 |
+------------------+----------------------+-------------+
5 rows in set (0.001 sec)

Query OK, 1 row affected (0.002 sec)

+---------------------+----------------------+---------------+
| charcter_name       | item_name            | item_quantity |
+---------------------+----------------------+---------------+
| Fennian Brightstone | Frostfire Blade      |             2 |
| Fennian Brightstone | Ring of Shadows      |             3 |
| Fennian Brightstone | Cloak of the Phoenix |             3 |
| Fennian Brightstone | Boots of the Cat     |             1 |
| Fennian Brightstone | Shield of Resilience |             3 |
+---------------------+----------------------+---------------+
5 rows in set (0.001 sec)

Query OK, 1 row affected (0.002 sec)

+------------------+----------------------+
| charcter_name    | ability_name         |
+------------------+----------------------+
| Xandril the Wise | Healing Aura         |
| Xandril the Wise | Counterspell         |
| Xandril the Wise | Inspirational Speech |
| Xandril the Wise | Frostbite            |
| Xandril the Wise | Shadow Step          |
+------------------+----------------------+
5 rows in set (0.001 sec)
